client/network/src/service.rs

```diff
@@ -259,6 +265,7 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 			params.block_announce_validator,
 			params.metrics_registry.as_ref(),
 			boot_node_ids.clone(),
+			metrics.as_ref().map(|m| m.notifications_queues_size.clone()),
```

Comment:
Let's say we would like a `Protocol` to expose more metrics in the future. In that case wouldn't it make more sense to passs a `&Registry` down to `Protocol::new` and have it register its own `Metrics` struct?

This is a preference, not a requirement.

Reply:
My general idea about this code is that is would eventually completely disappear after #5481. I'm trying as much as possible to be fully predictable/deterministic and report everything that is worth notifying to the parent.